src/Features/Core/Portable/ReplaceMethodWithProperty/ReplaceMethodWithPropertyCodeRefactoringProvider.cs

```diff
@@ -130,7 +130,7 @@ private static bool OverridesMethodFromSystemObject(IMethodSymbol method)
         {
             for (var current = method; current != null; current = current.OverriddenMethod)
             {
-                if (current.ContainingType.SpecialType == SpecialType.System_Object)
+                if (current.ContainingType.DeclaringSyntaxReferences.IsEmpty)
```

Comment:
This might be a costly check to do, as it was replaced in https://github.com/dotnet/roslyn/pull/23491/files

Reply:
It should not be costly.  You can keep it as before.

Reply:
💭 Yes, this is a relatively costly property. However, it should not be problematic considering this is a refactoring in a single context and not an analyzer that runs on the entire project.